From 222eedb93d505ce966e4ec7491edfc34080f668f Mon Sep 17 00:00:00 2001
From: Adrian Reber <areber@redhat.com>
Date: Tue, 15 Feb 2022 17:01:32 +0000
Subject: [PATCH 136/245] test: remove test for LOCK_MAND flock

Linux Kernel release 5.16 removed support for LOCK_MAND flock and so the
test to verify if LOCK_MAND works started to fail with 5.16.

The kernel also logs following message:

 Attempt to set a LOCK_MAND lock via flock(2). This support has been removed and the request ignored.

This fixes CRIU CI using Fedora with 5.16.

See Linux Kernel commit 90f7d7a0d0d68623b5f7df5621a8d54d9518fcc4
 "locks: remove LOCK_MAND flock lock support"

Signed-off-by: Adrian Reber <areber@redhat.com>
---
 test/zdtm/static/file_locks01.c | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/test/zdtm/static/file_locks01.c b/test/zdtm/static/file_locks01.c
index 6c2e54ff4..beea171f5 100644
--- a/test/zdtm/static/file_locks01.c
+++ b/test/zdtm/static/file_locks01.c
@@ -159,7 +159,6 @@ int main(int argc, char **argv)
 
 	flock(fd_0, LOCK_SH);
 	flock(fd_1, LOCK_EX);
-	flock(fd_2, LOCK_MAND | LOCK_READ);
 
 	test_daemon();
 	test_waitsig();
@@ -172,11 +171,6 @@ int main(int argc, char **argv)
 		fail("Failed on fd %d", fd_1);
 		ret |= 1;
 	}
-	if (check_file_lock(fd_2, "MSNFS", "READ", dev, inodes[2])) {
-		fail("Failed on fd %d", fd_2);
-		ret |= 1;
-	}
-
 	if (!ret)
 		pass();
 
-- 
2.35.1

